adaptive_components/lib/src/adaptive_column.dart

```diff
@@ -0,0 +1,131 @@
+        final double _margin = margin ?? _entry.margin;
+        final double _gutter = gutter ?? _entry.gutter;
+        final int _totalColumns = _entry.columns;
+
+        return Container(
+          width: MediaQuery.of(context).size.width - (_margin * 2),
+          margin: EdgeInsets.symmetric(horizontal: _margin),
+          constraints: BoxConstraints(
+            minWidth: _entry.adaptiveWindowType.widthRangeValues.start,
+            maxWidth: _entry.adaptiveWindowType.widthRangeValues.end,
+          ),
+          child: Wrap(
+            runSpacing: 8.0,
+            children: () {
+              int currentColumns = 0;
+              int totalGutters = 0;
+              List<Widget> children = [];
+              List<AdaptiveContainer> row = [];
+
+              for (AdaptiveContainer child in this.children) {
+                // The if statement checks if the adaptiveContainer child fits
+                // within the adaptive constraints.
+                if (child.adaptiveConstraints
+                    .withinAdaptiveConstraint(context)) {
+                  row.add(child);
+                  currentColumns += child.adaptiveColumn;
+
+                  if (currentColumns < _totalColumns) {
+                    totalGutters++;
+                  } else {
+                    int gutters = 0;
+                    if (currentColumns > _totalColumns) {
+                      totalGutters--;
+                    }
+
+                    for (AdaptiveContainer rowItem in row) {
+                      // maxWidth equals column width without margin and gutters
+                      // divided by the total number of adaptive containers.
+                      double maxWidth = (MediaQuery.of(context).size.width -
+                              _margin * 2 -
+                              _gutter *
+                                  (gutters == totalGutters &&
+                                          currentColumns > _totalColumns
+                                      ? 0
+                                      : totalGutters)) /
+                          _totalColumns *
+                          rowItem.adaptiveColumn;
+                      children.add(
+                        ConstrainedBox(
+                          constraints: BoxConstraints(
+                            minWidth: maxWidth,
+                            maxWidth: maxWidth,
+                          ),
+                          child: rowItem,
+                        ),
+                      );
+
+                      if (gutters < totalGutters && 1 < row.length) {
+                        children.add(
+                          SizedBox(
+                            width: _gutter,
+                            child: Container(
+                              color: Colors.grey,
+                            ),
+                          ),
+                        );
+                        gutters++;
```

Comment:
After this is incremented, where is it used?

Reply:
gutters now rowGutters represents the amount of gutters in the row currently being added. 

While totalGutters represents all the amount of gutters to expect to see in that row. Once rowGutters is equal to totalGutters than we don't need to add any more gutters